examples/04_glm_first_level/plot_two_runs_model.py

```diff
@@ -0,0 +1,258 @@
+# of the resulting brain maps.
+
+# %%
+# Compute the fixed effects statistics
+# using the preprocessed data of both runs.
+#
+# A more straightforward alternative to fitting run-specific GLMs,
+# than combining the results with :func:`~nilearn.glm.compute_fixed_effects`,
+# is to simply fit the GLM to both runs at once.
+#
+# Since we can assume that the design matrices of both runs
+# have the same columns, in the same order,
+# we can again reuse the first run's contrast vector.
+fmri_glm_multirun = fmri_glm.fit(fmri_imgs, design_matrices=design_matrices)
+
+# We can just define the contrast array for one run and assume
+# that the design matrix is the same for the other.
+# However, if we want to be safe, we should define each contrast separately,
+# and provide it as a list.
+contrast_val = [
+    np.array([[-1, -1, 1, 1]]),  # run 1
+    np.array([[-1, -1, 1, 1]]),  # run 2
+]
+
+z_map = fmri_glm_multirun.compute_contrast(
+    contrast_val,
+    output_type="z_score",
+)
+plotting.plot_stat_map(
+    z_map,
+    bg_img=mean_img_,
+    threshold=3.0,
+    cut_coords=cut_coords,
+    title=f"{contrast_id}, fixed effects",
+)
+
+plotting.show()
+
+# %%
+# You may note that the results are the same as the first fixed effects
+# analysis, but with a lot less code.
+
+# %%
+# Compute a range of contrasts across both runs
+# ---------------------------------------------
+# It may be useful to investigate a number of contrasts.
+# Therefore, we will move beyond the original contrast of interest
+# and both define and compute several.
+
+# %%
+# Contrast specification
+n_columns = design_matrices[0].shape[1]
+contrasts = {
+    "SStSSp_minus_DStDSp": np.array([[1, 0, 0, -1]]),
+    "DStDSp_minus_SStSSp": np.array([[-1, 0, 0, 1]]),
+    "DSt_minus_SSt": np.array([[-1, -1, 1, 1]]),
+    "DSp_minus_SSp": np.array([[-1, 1, -1, 1]]),
+    "DSt_minus_SSt_for_DSp": np.array([[0, -1, 0, 1]]),
+    "DSp_minus_SSp_for_DSt": np.array([[0, 0, -1, 1]]),
+    "Deactivation": np.array([[-1, -1, -1, -1, 4]]),
+    "Effects_of_interest": np.eye(n_columns)[:5, :],  # An F-contrast
+}
+
+# %%
+# Next, we compute and plot the statistics for these new contrasts.
+
+print("Computing contrasts...")
+for index, (contrast_id, contrast_val) in enumerate(contrasts.items()):
+    print(f"  Contrast {index + 1:02g} out of {len(contrasts)}: {contrast_id}")
+    # Estimate the contasts.
+    z_map = fmri_glm.compute_contrast(contrast_val, output_type="z_score")
+
+    # Write the resulting stat images to file.
+    z_image_path = output_dir / f"{contrast_id}_z_map.nii.gz"
+    z_map.to_filename(z_image_path)
+
+# %%
+# Generating a report
+# -------------------
+# Since we have already computed the FirstLevelModel
+# and have a number of contrasts,
+# we can quickly create a summary report.
+from nilearn.reporting import make_glm_report
+
+report = make_glm_report(
+    fmri_glm_multirun,
+    contrasts,
+    bg_img=mean_img_,
+    title="two-runs fMRI model fitting",
+)
```

Comment:
should we showcase `glm.generate_report` method here? Is there a reason to favor using the function over the class method?

Reply:
not that I know of
makes sense using both as in any case the method is just an indirect call to the function